pkg/netpol/diff/diff.go

```diff
@@ -79,31 +79,42 @@ func (d diffMap) update(key string, isFirst bool, c connlist.Peer2PeerConnection
 // TODO: should modify the keys for ip-blocks, should work with disjoint ip-blocks from both results .
 func diffConnectionsLists(conns1, conns2 []connlist.Peer2PeerConnection) (ConnectivityDiff, error) {
 	// convert to a map from src-dst full name, to its connections pair (conns1, conns2)
-	diffMap := diffMap{}
+	diffsMap := diffMap{}
 	for _, c := range conns1 {
-		diffMap.update(getKeyFromP2PConn(c), true, c)
+		diffsMap.update(getKeyFromP2PConn(c), true, c)
 	}
 	for _, c := range conns2 {
-		diffMap.update(getKeyFromP2PConn(c), false, c)
+		diffsMap.update(getKeyFromP2PConn(c), false, c)
 	}
 	res := &connectivityDiff{
 		removedConns: []connlist.Peer2PeerConnection{},
 		addedConns:   []connlist.Peer2PeerConnection{},
 		changedConns: []*connsPair{},
 	}
-	for _, d := range diffMap {
-		if d.firstConn != nil && d.secondConn != nil {
-			res.changedConns = append(res.changedConns, d)
-		} else if d.firstConn != nil {
+	for _, d := range diffsMap {
+		switch {
+		case d.firstConn != nil && d.secondConn != nil:
+			if !equalConns(d.firstConn, d.secondConn) {
+				res.changedConns = append(res.changedConns, d)
+			}
+		case d.firstConn != nil:
 			res.removedConns = append(res.removedConns, d.firstConn)
-		} else if d.secondConn != nil {
+		case d.secondConn != nil:
 			res.addedConns = append(res.addedConns, d.secondConn)
+		default:
+			continue
 		}
 	}
 
 	return res, nil
 }
 
+// checks whether two connlist.Peer2PeerConnection objects are equal
+func equalConns(firstConn, secondConn connlist.Peer2PeerConnection) bool {
+	return firstConn.AllProtocolsAndPorts() == secondConn.AllProtocolsAndPorts() &&
+		reflect.DeepEqual(firstConn.ProtocolsAndPorts(), secondConn.ProtocolsAndPorts())
+}
+
```

Comment:
can you convert to `common.ConnectionSet` objects and use its `Equals` method instead?

Reply:
it is possible, but we'll need to wait until merging the "ingress_analyzer" PRs , since `common` is not merged into main branch yet

Reply:
committed changes to #180.
closing this PR